### DISCUSSION TITLE
[Feature request] Descew on multiple lines of reference

### DISCUSSION THREAD

**Discussion by juscghwe** [ScanTailor-Advanced/scantailor-advanced#71] [Ideas]
I think it would be great if we had an option to add several lines of reference in the manual adjustments. 

_User story:
To undo the book fold I (and most of my coworkers) press down the page. This leads to 'smoother' scans as the left/right page can completely be seen and lines most of the time are clear. But we also cause several new folding lines: The one on the top from the book folding itself (concav) and a new convex line from the bottom. Sometimes these overlap, forming medium complex nets._

I made a screenshot of the resulting net from the two reference lines and highlighted the line where the net is most off. This is not the best example but I dont have a more 'extreme' at hand right now (screenshot on the bottom).
From my perspective, adding an **optional reference line in the middle** would solve this issue in most cases.

<img width="589" alt="Screenshot 2023-08-01 153913" src="https://github.com/4lex4/scantailor-advanced/assets/62502225/787986d6-67f9-46ba-b6db-b3ab49fed31e">

Also at [https://imgur.com/a/ZZKNdTm](https://imgur.com/a/ZZKNdTm)

**Comment by zvezdochiot** [comment]
Hi @juscghwe .

:question: Those. do you want to split one grid into two that share the same line?

It is possible to split the mesh, but it will be very difficult to implement a composite transformation. And such an implementation will be full of bugs and UB's.

**Reply by juscghwe** [reply]
I rather thought of 'adding weight' to certain areas of the mesh as the problem seems to occure due to the homogenic 30/30 mesh. But I get your point as I can't come up with a 'fast' equasion to add a third vertical curve to the mesh without cropping it into two field. And the output of two meshes would hardly depend on the user putting the third curve exactly at .5 height.

At the moment I'm writing my masters exam but I'll try to figure a mathematical approach as soon as I finished my paper. I'll keep you updated.
